bats/tests/profile/sample.bats

```diff
@@ -32,69 +45,188 @@ start_app() {
     wait_for_container_engine
 }
 
-verify_settings() {
+verify_profiles() {
     PROFILE_TYPE=$PROFILE_LOCKED
     run profile_exists
-    "${assert}_success"
+    "${assert}_success" || return
 
     PROFILE_TYPE=$PROFILE_DEFAULTS
     run profile_exists
-    "${assert}_success"
+    "${assert}_success" || return
+}
 
+verify_settings() {
+    #settings from defaults profile
     run get_setting .containerEngine.name
-    "${assert}_output" "$PROFILE_CONTAINER_ENGINE"
+    "${assert}_output" "$PROFILE_CONTAINER_ENGINE" || return
 
     run get_setting .application.startInBackground
-    "${assert}_output" "$PROFILE_START_IN_BACKGROUND"
+    "${assert}_output" "$PROFILE_START_IN_BACKGROUND" || return
+    #settings from locked profile
+    run get_setting .containerEngine.allowedImages.enabled
+    "${assert}_output" "$PROFILE_USE_IMAGE_ALLOW_LIST" || return
+
+    run get_setting .containerEngine.allowedImages.patterns
+    "${assert}_output" --partial "${PROFILE_IMAGE_PATTERNS[@]}" || return
+
+    run get_setting .application.extensions.allowed.enabled
+    "${assert}_output" "$PROFILE_USE_EXTENSION_LIST" || return
+
+    run get_setting .application.extensions.allowed.list
+    "${assert}_output" --partial "${PROFILE_EXTENSION_LIST[@]}" || return
+
+    run get_setting .kubernetes.version
+    "${assert}_output" "$PROFILE_LOCKED_KUBERNETES_VERSION" || return
+    refute_output "$PROFILE_DEFAULTS_KUBERNETES_VERSION" || return
 }
 
 @test 'initial factory reset' {
     factory_reset
 }
 
-@test 'start up without profile' {
+@test 'start up with NO profiles' {
     RD_USE_PROFILE=false
-    start_app
+    RD_USE_IMAGE_ALLOW_LIST=false
+    start_application
 }
 
-@test 'verify default settings' {
+@test 'verify there are NO profiles created' {
+    before verify_profiles
+}
+
+@test 'verify default settings were applied' {
     before verify_settings
 }
 
-@test 'factory reset' {
+@test 'factory reset before creating profiles' {
     factory_reset
 }
 
-@test 'create profile' {
+@test 'create profiles' {
     PROFILE_TYPE=$PROFILE_LOCKED
     create_profile
-    add_profile_string containerEngine.name "$PROFILE_CONTAINER_ENGINE"
 
     PROFILE_TYPE=$PROFILE_DEFAULTS
     create_profile
+
+    verify_profiles
```

Comment:
On second thoughts, we should fix this on Windows by creating an empty key at the profile location after deleting the subtree. That would be more consistent with how it works on the other platforms. I can create a PR to implement this later.

Reply:
I think this should do it; maybe just add it to your PR and test it. In `create_profile` after these lines:

```shell
        # Make sure any old profile data at this location is removed
        run profile_reg delete "."
```

add

```shell
        run profile_reg add "."
```

I haven't tested it (on macOS right now), but I'm 95% sure this will work and makes sure the `profile_exists` succeeds after running `create_profile`.